code/modules/projectiles/guns/energy/energy_gun.dm

```diff
@@ -72,6 +72,29 @@
 	ammo_type = list(/obj/item/ammo_casing/energy/net, /obj/item/ammo_casing/energy/trap)
 	can_flashlight = 0
 	ammo_x_offset = 1
+	var/obj/item/device/radio/beacon/guntarget
+
+/obj/item/gun/energy/e_gun/dragnet/proc/set_target(target, user)//called on beacon's attackby()
+	guntarget = target
+	to_chat(user, "<span class='notice'>[src] has locked onto [guntarget].</span>")
+	//qdel(chambered)
```

Comment:
i put this and the line below it here because I'm unsure of where ammo casings are spawned (i think multiple procs spawn them if they're missing which is irritating) so i assumed that if the ammo casings spawned with the gun and then you were to set a tele target, the existing casings wouldn't update. 

i have commented it out because there are a lot of bugs right now and i need to simplify things